lib/src/supertokens-http-client.dart

```diff
@@ -281,10 +284,39 @@ class Client extends http.BaseClient {
         await SuperTokensUtils.getLocalSessionState();
     if (preRequestLocalSessionState.status ==
         LocalSessionStateStatus.NOT_EXISTS) {
-      AntiCSRF.removeToken();
-      FrontToken.removeToken();
+      await AntiCSRF.removeToken();
+      await FrontToken.removeToken();
     }
   }
+
+  static Map<String, dynamic> _generateMapFromCookie(String cookie) {
+    if (cookie.isEmpty) return {};
+    List<String> entries = cookie.split(";");
+    Map<String, dynamic> map = {};
+    entries.forEach((element) {
+      List<String> keyValuePair = element.split("=");
+      if (keyValuePair.length == 2) {
+        map[keyValuePair[0]] = keyValuePair[1];
+      }
+    });
+    return map;
+  }
+
+  static String _generateCookieHeader(String? oldCookie, String? newCookie) {
+    Map<String, dynamic> existingCookieHeaderMap =
+        _generateMapFromCookie(oldCookie ?? "");
+    Map<String, dynamic> newCookiesToAddMap =
+        _generateMapFromCookie(newCookie ?? "");
+    String finalCookieHeader = "";
+    existingCookieHeaderMap.keys.forEach((element) {
+      if (newCookiesToAddMap.containsKey(element)) {
+        finalCookieHeader += "$element=${newCookiesToAddMap[element]};";
+      } else {
+        finalCookieHeader += "$element=${existingCookieHeaderMap[element]};";
+      }
+    });
+    return "${finalCookieHeader}HttpOnly;";
+  }
```

Comment:
Why not do:

1. filter existing cookies in a way that no cookies with the same names exist in newCookiesToAdd
2. Combine newCookiesToAdd and the filtered cookies
3. join with ";"